crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -629,6 +629,132 @@ impl Torus {
+    #[inline(always)]
+    pub fn new(a: Vec3, b: Vec3, c: Vec3) -> Self {
+        Self {
+            vertices: [a, b, c],
+        }
+    }
+
+    /// Get the area of the triangle
+    #[inline(always)]
+    pub fn area(&self) -> f32 {
+        let [a, b, c] = self.vertices;
+        let ab = a.distance(b);
+        let bc = b.distance(c);
+        let ca = c.distance(a);
+
+        let s = (ab + bc + ca) / 2.0;
+        (s * (s - ab) * (s - bc) * (s - ca)).sqrt()
+    }
+
+    /// Get the perimeter of the triangle
+    #[inline(always)]
+    pub fn perimeter(&self) -> f32 {
+        let [a, b, c] = self.vertices;
+        let ab = a.distance(b);
+        let bc = b.distance(c);
+        let ca = c.distance(a);
+        ab + bc + ca
+    }
+
+    /// Get the normal of the triangle in the direction of the right-hand rule, assuming
+    /// the vertices are ordered in a counter-clockwise direction.
+    ///
+    /// The normal is computed as the cross product of the vectors `ab` and `ac`.
+    #[inline(always)]
+    pub fn normal(&self) -> Dir3 {
+        let [a, b, c] = self.vertices;
+        let ab = b - a;
+        let ac = c - a;
+        Dir3::new_unchecked(ab.cross(ac).normalize())
+    }
+
+    /// Reverse the order of the vertices in the triangle
+    #[inline(always)]
+    pub fn reverse(&mut self) {
+        self.vertices.swap(1, 2);
+    }
+
+    /// Get the centroid of the triangle
+    #[inline(always)]
+    pub fn centroid(&self) -> Vec3 {
+        (self.vertices[0] + self.vertices[1] + self.vertices[2]) / 3.0
+    }
+}
+
+impl Bounded3d for Triangle3d {
+    /// Get the bounding box of the triangle
+    fn aabb_3d(&self, translation: Vec3, rotation: glam::Quat) -> crate::bounding::Aabb3d {
+        let [a, b, c] = self.vertices;
+        let a = rotation * a + translation;
+        let b = rotation * b + translation;
+        let c = rotation * c + translation;
+
+        let min_x = a.x.min(b.x).min(c.x);
+        let min_y = a.y.min(b.y).min(c.y);
+        let min_z = a.z.min(b.z).min(c.z);
+        let max_x = a.x.max(b.x).max(c.x);
+        let max_y = a.y.max(b.y).max(c.y);
+        let max_z = a.z.max(b.z).max(c.z);
+
+        let bounding_center = Vec3::new(
+            (min_x + max_x) / 2.0,
+            (min_y + max_y) / 2.0,
+            (min_z + max_z) / 2.0,
+        );
+
+        let half_extents = Vec3::new(
+            (max_x - min_x) / 2.0,
+            (max_y - min_y) / 2.0,
+            (max_z - min_z) / 2.0,
+        );
+
+        crate::bounding::Aabb3d::new(bounding_center, half_extents)
+    }
+
+    /// Get the bounding sphere of the triangle
+    fn bounding_sphere(
+        &self,
+        translation: Vec3,
+        rotation: glam::Quat,
+    ) -> crate::bounding::BoundingSphere {
+        let [a, b, c] = self.vertices;
+        let a = rotation * a + translation;
+        let b = rotation * b + translation;
+        let c = rotation * c + translation;
```

Comment:
You can skip doing these 3 transformations and instead only transform the resulting center.

The reason is that spheres are rotationally symmetric, and rotations and translations are isometries, meaning they preserve distances. (Therefore the center - vertex distances are the same before and after transforming. And the center-vertex distances are what is important for a bounding sphere.)